services/grafana/client_test.go

```diff
@@ -125,16 +125,19 @@ func TestClient(t *testing.T) {
 	})
 
 	t.Run("CreateAnnotation", func(t *testing.T) {
-		t.Skip("https://jira.percona.com/browse/PMM-3812")
-
 		from := time.Now()
 
+		req, err := http.NewRequest("GET", "/dummy", nil)
+		require.NoError(t, err)
+		req.SetBasicAuth("admin", "admin")
+		authorization := req.Header.Get("Authorization")
```

Comment:
should we write test for wrong auth? I know that it's almost impossible case

Reply:
Such a test case looks like a try to cover not our code - auth checks are on a higher level.
And here, we can only test if we receive the expected error from Grafana.
An additional auth error interceptor is only required here if we need to change the error message. Is there any reason to do it?

Reply:
ok, let's just add Annotation endpoint to auth_server_test.go then to be sure that it secures us.
https://github.com/percona/pmm-managed/blob/PMM-2.0/services/grafana/auth_server_test.go#L191

Reply:
Your suggestion will not work as that method support only GET.
BTW, added a test with incorrect creds.

Reply:
I'm not sure that it matters, but if you think let's write test for POST then :) to be sure that

Reply:
> I'm not sure that it matters, but if you think let's write test for POST then :) to be sure that
It matters - because GET can do viewer, but to add annotation with POST user should be at least editor.

Reply:
What @BupycHuk probably wanted to say and what should be done:
`/management.Annotation/AddAnnotation` API is available only to admin user thanks to this line: https://github.com/percona/pmm-managed/blob/45edb31bfcdcd539a6953e93cdd0cd8b0b2c9385/services/grafana/auth_server.go#L40
It makes sense to add an explicit test for it there: https://github.com/percona/pmm-managed/blob/45edb31bfcdcd539a6953e93cdd0cd8b0b2c9385/services/grafana/auth_server_test.go#L194-L208

Reply:
What Grafana uses for their API call is not relevant

Reply:
Yeah, I just want ensure that we always get `Authorization` header and it's always admin, so I want to add new line to the test. It's about pmm-managed API, not Grafana.

Reply:
Oh, now I understand. Thank you for your good explanation. Added.